core/data-transformers/product-card-transformer.ts

```diff
@@ -7,9 +8,43 @@ import { ProductCardFragment } from '~/components/product-card/fragment';
 
 import { pricesTransformer } from './prices-transformer';
 
+const getInventoryMessage = (
+  product: ResultOf<typeof ProductCardFragment>,
+  outOfStockMessage?: string,
+  showBackorderMessage?: boolean,
+) => {
+  if (!product.inventory.isInStock) {
+    return outOfStockMessage;
+  }
+
+  if (!showBackorderMessage || product.inventory.hasVariantInventory) {
+    return undefined;
+  }
+
+  const { availableForBackorder, unlimitedBackorder } = product.inventory.aggregated || {};
+
+  const hasBackorderAvailablity = !!availableForBackorder || unlimitedBackorder;
+
+  if (!hasBackorderAvailablity) {
+    return undefined;
+  }
```

Comment:
Why isn't `showBackorderMessage` coming with a value of `true` or `false` from the BE without having to do these checks in the FE? Any reasoning behind it that you know of?